### ISSUE TITLE
counting error

### ISSUE THREAD

**Issue by ChunmianLin** [ahmetozlu/tensorflow_object_counting_api#44]
hi, thanks for your great job! I successfully ran the script 'vehicle_counting.py', but when I check the result 'output.avi', it seems some errors happened:
1. vehicle counting: in the resulting video, the value of vehicle through the roi_position cannot match with vehicle counting that presents in the video.
2. error bbox: in the resulting video, there are some blue bounding box labelled 'smurf', which makes me so confused that what is the problems?

Can you help me fix this problems? Thanks a lot!

**Comment by ahmetozlu** [comment]
1. It depends the object detector, when the object detector could not detect the object then the object counting algorithm can not work properly. I suggest you try a different object detector (YOLO, RCNN, etc.) thus there must not be any false positive or true negative counted objects.
The SSD MobileNet detector in this repo is a general purpose one (not optimized specifically for vehicle/pedestrian detection) thus there can be some missed detection so it effects the counting algorithm performance directly.
2. It is not possible, are you sure that you use SSD MobileNet frozen inference graph and SSD MobileNet label map? It seems like you use wrong model or label map.

**Comment by ChunmianLin** [comment]
@ahmetozlu Thanks.
1. I have try to other pretrained models downloaded from tensorflow/models website, like SSD_resnet, Faster RCNN_resnet, but the performance is not substantially better. So I don't check where the problem and cannot solve it thoroughtly.
2. And then, I am not very sure whether I loaded wrong model and label map. My setting follows your default condition and make a little modification. The resulting video predicts some bounding box with 'smurf', as follows. How can I tackle with?
![2019-08-16 09-58-51屏幕截图](https://user-images.githubusercontent.com/53289159/63138391-ea9fea00-c00c-11e9-853b-0b1a7ae720bb.png)
![2019-08-16 10-00-01屏幕截图](https://user-images.githubusercontent.com/53289159/63138394-ed024400-c00c-11e9-8e90-4dc0e3233eb6.png)
what's more, I find that the value of vehicle passing through the ROI position cannot match to the counting, but I don't know how to solve it. I really need your help. Thanks in advance!

**Comment by ahmetozlu** [comment]
1. It is fixed, can you check again, please?
2. Since the SSD MobileNet detector in this repo is a general purpose one (not optimized specifically for vehicle/pedestrian detection) thus there can be some false positives or false negatives such as "surfboard" detection on the frame you shared. You can consider about performing transfer learning to increase the accuracy.

**Comment by ShaoDu** [comment]
@ChunmianLin Hello, I have the same problem. Can we discuss it privately

**Comment by ChunmianLin** [comment]
@ahmetozlu Thanks.
1. I am testing for it.
2. I don't think this is problem of SSD Mobilenet or the insufficient accuracy. As I mentioned above, I have try other pretrained model, but the error still exists. The label map is corresponding to pretrained model, and test data is traffic survaillance. It seems to be absurd that geenrates results labelled 'smurf'? Could you please check your code that is shared with other applications? I think maybe there exist some errors in some shared feature layers or operations. 
Thanks a lot in advance.

**Comment by ChunmianLin** [comment]
@ShaoDu Sure. How to contact with you privately?

**Comment by ShaoDu** [comment]
@ChunmianLin Are you Chinese? If so, how can we communicate

**Comment by ChunmianLin** [comment]
yes.@ShaoDu

**Comment by ShaoDu** [comment]
@ChunmianLin my qq:2577414131. Can you add me

**Comment by ahmetozlu** [comment]
This issue totally solved, please try to run [vehicle_counting.py](https://github.com/ahmetozlu/tensorflow_object_counting_api/blob/master/vehicle_counting.py) and check the result.
